ci-operator/step-registry/mirror-images/by-oc-adm/mirror-images-by-oc-adm-commands.sh

```diff
@@ -20,15 +20,21 @@ if [ ! -f "${SHARED_DIR}/mirror_registry_url" ]; then
 fi
 echo "MIRROR_REGISTRY_HOST: $MIRROR_REGISTRY_HOST"
 
+if [[ -n "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" ]]; then
```

Comment:
> The [[ X"$RELEASE_IMAGE_LATEST" != X"$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" ]] is a redundant check, which is not needed.
You can check with "OVERRIDE=aaa, LATEST=bbb", and "OVERRIDE=ccc, LATEST=ccc"

Hello,

We were discussing of adding that further check (the inequality) to stay safe and because the first one (non empty string) seems always satisfied: e.g., the install step will fail if OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE is empty https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L248-L252

@liangxia if you are sure they are always the same, shall we remove the if at all and only use "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"?

Reply:
I prefer to keep checking `[[ -n "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" ]]`, the logic looks good to me.

Reply:
Hi @jianlinliu, ok, but...

If we understand/agree on this to also be a redundancy, some other changes that will be needed only to enable testing on arm64 could be avoided. 

The changes from this PR would only be to fix the use of "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" instead of "$RELEASE_IMAGE_LATEST". If the above is true, the installer team, as well as any other guy doing these PRs could use "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" in place of "$RELEASE_IMAGE_LATEST" and there is no need for further debugging by the arm64 team when installations don't work due to this reason. The result would be a more straightforward multi-arch support for the jobs.

What I'm saying is just that is helpful to understand where we should use one or the other variable.

Reply:
Some of the installation only provides RELEASE_IMAGE_LATEST, that's why `[[ -n "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" ]]` is still needed.

Reply:
RELEASE_IMAGE_LATEST is provided by the ci-operator and is set to be the pullspec corresponding to "release:latest". Nobody overrides it in the current /config.

OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE, in this job, is set to the same value by default (in the yaml spec of the step below). It will never be empty unless someone forces it (and this is a user error that should cause a bunch of other errors down the road).

The ci-operator will fill both the variables when executing the mirroring to the same value.

Is there a case where this variable will be filled with an empty value in this step?

https://github.com/openshift/release/blob/ae170644a22bc6ccf21de1b1a0ab05b148c0d86d/ci-operator/step-registry/mirror-images/by-oc-adm/mirror-images-by-oc-adm-ref.yaml#L15-L19

Reply:
yes, the OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE will never be empty if set dependencies in steps, got an example [/ipi-install-install-aws-ref.yaml](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/aws/ipi-install-install-aws-ref.yaml#L18-L22) and [ipi-install-install-aws-commands.sh](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/aws/ipi-install-install-aws-commands.sh#L65-L68)

Reply:
@aleskandro I know what you mean here, yes, we do not need to check `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` in this case.

Reply:
One interesting thing is,  `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` is more straightforward for multi-arch team, `RELEASE_IMAGE_LATEST` is more straightforward for installer (x86_64)

Reply:
@yunjiang29, the issue is that RELEASE_IMAGE_LATEST should be the right one to use (similar to the  installer_payload_image that we have in flexy), and I agree. It is automatically provided by the ci-operator based on the values in the `releases` stanza of the config yaml.

However, this dependency is used by many other steps throughout this repo, and until we (and MA eng.) will fix all the steps to be multi-arched and/or run on clusters like the [arm01](https://github.com/openshift/release/tree/master/clusters/build-clusters/arm01), it seems hard to use "RELEASE_IMAGE_LATEST".

Also note that the installer team also uses OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE in the main installer steps, both [ipi](https://github.com/openshift/release/tree/master/ci-operator/step-registry/ipi) and [upi](https://github.com/openshift/release/tree/master/ci-operator/step-registry/upi).